### ISSUE TITLE
Edge Mask not being used in GNNExplainer

### ISSUE THREAD

**Issue by danilonumeroso** [rusty1s/pytorch_geometric#1320]
## 🐛 Bug

I looked at the implementation, this is the training cycle: 
![image](https://user-images.githubusercontent.com/26821153/84590682-f0d34280-ae38-11ea-9f8d-17d80d10f926.png)

When predicting the class we always pass edge_index to the model, without ever
modifying it. 

## Expected behavior
The learnt edge mask M should multiply the sparse adjacency matrix, as the original paper does:

![image](https://user-images.githubusercontent.com/26821153/84590803-e6657880-ae39-11ea-957c-2e65f14604cc.png)

**Comment by rusty1s** [comment]
The `edge_mask ` in injected into each `MessagePassing` layer, see [here](https://github.com/rusty1s/pytorch_geometric/blob/master/torch_geometric/nn/conv/message_passing.py#L312-L321). That allows us to use *any* GNN operator in conjunction with `GNNExplainer`.

**Comment by danilonumeroso** [comment]
I knew I was missing something, thanks for the explanation
